mmv1/third_party/terraform/resources/resource_sql_user.go

```diff
@@ -236,7 +236,10 @@ func resourceSqlUserCreate(d *schema.ResourceData, meta interface{}) error {
 		pp := expandPasswordPolicy(v)
 		user.PasswordPolicy = pp
 	}
-
+	err = validateUserType(typ, host, password)
+	if err != nil {
+		return err
+	}
```

Comment:
Could this work be pulled out into a separate file? I'm not sure how this validation links to adding the new data source